src/machinery/WalletState.ts

```diff
@@ -0,0 +1,11 @@
+import type { NanoAddress, NanoWallet } from './models';
+import { walletStore } from '../stores/stores';
+
+export interface WalletState {
+  wallet: NanoWallet | undefined;
+  selectedAccount: NanoAddress | undefined;
+}
+
+export function setWalletState(walletState: WalletState) {
```

Comment:
I like this, keeping domain + helper functions together like this. Perhaps we should move the creating of stores in to each file as well?